mule-extensions-api/src/main/java/org/mule/runtime/extension/api/util/ExtensionModelUtils.java

```diff
@@ -268,6 +270,39 @@ private void resolve(ComponentModel model, Object owner) {
     return result.get();
   }
 
+  /**
+   * @param extensionModel the model which owns the {@code component}
+   * @param component      a component
+   * @return A {@link Set} with the {@link ConfigurationModel} that the can be used alongside with the {@code component}
+   */
+  public static Set<ConfigurationModel> getConfigurationForComponent(ExtensionModel extensionModel, ComponentModel component) {
+    Set<ConfigurationModel> result = new HashSet<>();
+    new ExtensionWalker() {
```

Comment:
very inneficint to use a walker for this use case. Why not simply iterate over extensionModel.getConfigurtionModel() ?